src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -157,6 +157,7 @@ internal static IPHostEntry InternalGetHostByAddress(IPAddress address, bool inc
         public static string GetHostName()
         {
             GlobalLog.Print("Dns.GetHostName");
+            NameResolutionPal.EnsureSocketsAreInitialized();
```

Comment:
Out of curiosity, why move these calls out of the PAL? Personally, I'd leave it up to the PAL implementation to check for any required initialization and leave it out of the up-stack code.